exporters/elasticsearch/include/opentelemetry/exporters/elasticsearch/es_log_recordable.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <map>
+#include <unordered_map>
+#include "nlohmann/json.hpp"
+#include "opentelemetry/sdk/common/attribute_utils.h"  // same as traces/attribute_utils
+#include "opentelemetry/sdk/logs/recordable.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+
+/**
+ * An Elasticsearch Recordable implemenation that stores the 10 fields of the Log Data Model inside
+ * a JSON object,
+ */
+class ElasticSearchRecordable final : public sdk::logs::Recordable
```

Comment:
nit: maybe rename `JSONLogRecord`? Could maybe be used for other exporters?

Reply:
Don't think we can use it for others, as json format here would be specific to Elastic. Current name looks good in that case.

Reply:
ok pls ignore my comment, I think there is no format defined for elastic search, and we are kind of using somewhat generic format here. Still I feel we can keep current name : )

Reply:
haha yeah, i havent got any strong opinions on it - I agree it's a pretty generic format. so I'm wondering whether there's more use cases for it - maybe logging structured JSON data to a file would be useful? so those files could be used to be imported to databases or something? Im good with the name either way :)